lab_2_retrieval_w_bm25/main.py

```diff
@@ -62,6 +95,22 @@ def calculate_tf(vocab: list[str], document_tokens: list[str]) -> dict[str, floa
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(vocab, list) or not all(isinstance(elem, str) for elem in vocab)\
+        or not isinstance(document_tokens, list) or not vocab or not document_tokens\
+            or not all(isinstance(elem, str) for elem in document_tokens):
+        return None
+
+    tf = {}
+    for stopword in document_tokens:
+        if stopword not in vocab:
+            tf[stopword] = document_tokens.count(stopword) / len(document_tokens)
+    for word in vocab:
+        tf[word] = 0.0
+        if word in document_tokens or tf[word] != 0.0:
+            tf[word] = document_tokens.count(word) / len(document_tokens)
+            continue
```

Comment:
why is the continue operator needed here?

Reply:
it's not needed, yes... fixed